signal/cosmos.go

```diff
@@ -52,6 +54,9 @@ func (c *Cosmos) getHeaders() map[string]string {
 }
 
 func (c *Cosmos) getEndpoints() []string {
+	if len(c.Endpoints) != 1 {
+		log.Errorf("Cosmos needs 1 endpoint, got %d", len(c.Endpoints))
```

Comment:
that is fine. but you can error out in the caller function and still return an error